plugins/qlcplus_4.12.1/presets.js

```diff
@@ -0,0 +1,1002 @@
+  },
+  PulseSlowToFast: {
+    isApplicable: cap => exportHelpers.isShutterEffect(cap, `Pulse`) && exportHelpers.isIncreasingSpeed(cap),
+    importCapability: () => importHelpers.getShutterStrobeCap(`Pulse`, `slow`, `fast`)
+  },
+  PulseFastToSlow: {
+    isApplicable: cap => exportHelpers.isShutterEffect(cap, `Pulse`) && exportHelpers.isDecreasingSpeed(cap),
+    importCapability: () => importHelpers.getShutterStrobeCap(`Pulse`, `fast`, `slow`)
+  },
+  RampUpSlowToFast: {
+    isApplicable: cap => exportHelpers.isShutterEffect(cap, `RampUp`) && exportHelpers.isIncreasingSpeed(cap),
+    importCapability: () => importHelpers.getShutterStrobeCap(`RampUp`, `slow`, `fast`)
+  },
+  RampUpFastToSlow: {
+    isApplicable: cap => exportHelpers.isShutterEffect(cap, `RampUp`) && exportHelpers.isDecreasingSpeed(cap),
+    importCapability: () => importHelpers.getShutterStrobeCap(`RampUp`, `fast`, `slow`)
+  },
+  RampDownSlowToFast: {
+    isApplicable: cap => exportHelpers.isShutterEffect(cap, `RampDown`) && exportHelpers.isIncreasingSpeed(cap),
+    importCapability: () => importHelpers.getShutterStrobeCap(`RampDown`, `slow`, `fast`)
+  },
+  RampDownFastToSlow: {
+    isApplicable: cap => exportHelpers.isShutterEffect(cap, `RampDown`) && exportHelpers.isDecreasingSpeed(cap),
+    importCapability: () => importHelpers.getShutterStrobeCap(`RampDown`, `fast`, `slow`)
+  },
+
+
+  // color capabilities
+
+  ColorMacro: {
+    isApplicable: cap => (cap.type === `ColorPreset` || cap.isSlotType(`Color`)) && cap.colors !== null && cap.colors.allColors.length === 1,
+    exportRes1: cap => cap.colors.allColors[0],
+    importCapability: capData => capabilityPresets.ColorDoubleMacro.importCapability(capData)
+  },
+  ColorDoubleMacro: {
+    isApplicable: cap => (cap.type === `ColorPreset` || cap.isSlotType(`Color`)) && cap.colors !== null && cap.colors.allColors.length === 2,
+    exportRes1: cap => cap.colors.allColors[0],
+    exportRes2: cap => cap.colors.allColors[1],
+    importCapability: ({ channelName, res1, res2, index }) => {
+      if (channelName.match(/wheel\b/i)) {
+        return {
+          type: `WheelSlot`,
+          slotNumber: index + 1
+        };
+      }
+
+      const colors = [res1];
+      if (res2) {
+        colors.push(res2);
+      }
+
+      return {
+        type: `ColorPreset`,
+        comment: ``,
+        colors
+      };
+    }
+  },
+  ColorWheelIndex: {
+    isApplicable: cap => cap.type === `WheelRotation` && cap.wheels[0].type === `Color` && exportHelpers.isRotationAngle(cap),
+    importCapability: () => ({
+      type: `WheelRotation`,
+      angleStart: `0deg`,
+      angleEnd: `360deg`,
+      helpWanted: `Are these the correct angles?`
+    })
+  },
+
+
+  // gobo capabilities
+
+  // TODO: import/export a gobo image as res1
+  GoboShakeMacro: {
+    isApplicable: cap => cap.type === `WheelShake` && (cap.isSlotType(/Gobo|Iris|Frost/) || cap.isSlotType(`Open`)),
```

Comment:
Why `` cap.isSlotType(`Open`) ``? Would this also match open color slots? And why not adding it to the Regex?

Reply:
Shaking open (color or gobo) slots can be best represented using the `GoboShakeMacro` and an image as res1. You're right about the regexp, though.